### ISSUE TITLE
Question about fine-tuning for text classification

### ISSUE THREAD

**Issue by yinhua1** [AnswerDotAI/ModernBERT#157]
Thank you for the exceptional work! I have a query concerning the fine-tuning process for text classification. 
I've been following the code provided in the 'examples/finetune_modernbert_on_glue.ipynb' notebook, but I've encountered some perplexing results. Could you please provide some guidance?
![屏幕截图 2024-12-23 235715](https://github.com/user-attachments/assets/182f2e85-47ac-4e1f-8637-b4529922cf63)

Here are more details:
1. I noticed that the var 'wd'  wasn't being utilized, so I included it for weight decay, but this has led to confusing results too. You can see the screenshot of the error here:
![屏幕截图 2024-12-24 000556](https://github.com/user-attachments/assets/51f7b0c5-cb15-4d83-958d-b9eca2287a57)
2.  I received a TypeError stating that ModernBertForSequenceClassification.__init__() received an unexpected keyword argument 'compile'. Here's the code snippet where the error occurred:
```python
hf_model = AutoModelForSequenceClassification.from_pretrained(
    checkpoint, num_labels=n_labels, id2label=id2label, label2id=label2id, compile=False
)
```
Could you please provide guidance on how to address these issues? Thank you in advance for your help!

**Comment by unnir** [comment]
the same issue.

**Comment by istoneyou** [comment]
same issue

**Comment by Quickstilver** [comment]
same issue here

**Comment by NohTow** [comment]
ping @ohmeow, if you could have a look

**Comment by wrq9** [comment]
same issue

**Comment by wrq9** [comment]
I have found the solution. Replace `complie=False` with `reference_compile=False`.

**Comment by hungnlp** [comment]
same issue when I finetune from my pretrain
